ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -274,17 +276,25 @@ untickedShelleyLedgerTipPoint ::
   -> Point (ShelleyBlock proto era)
 untickedShelleyLedgerTipPoint = shelleyTipToPoint . untickedShelleyLedgerTip
 
+validationMode :: Validation -> STS.ValidationPolicy
+validationMode ValidationOn  = STS.ValidateAll
+validationMode ValidationOff = STS.ValidateNone
```

Comment:
When I say we need to be able to use any validation level, I really mean any validation level that is provided by STS. This approach of redefining new validation types hides an important validation level `ValidateSuchThat`.
I suggest passing through the validation level provided by STS instead creating some isomorphic type, this will also ensure it will not go out of sync if there is a change in the validation level in STS.

Reply:
Please check the latest commit on this PR. That is from an outdated version.